[16.0][FIX] purchase_delivery_split_date

DavidJForgeFlow opened the pull request:
When the PO is already confirmed and you add a new line with a different date than the existing one, the new receipt is created but the old receipt and move have the dates updated:

<img width="1199" height="707" alt="image" src="https://github.com/user-attachments/assets/1d2bf852-0240-4e4a-a4ca-43322f7079bf" />
<img width="1219" height="676" alt="image" src="https://github.com/user-attachments/assets/1cea09b7-5e8c-426c-8b44-18fc3270f37c" />
<img width="1219" height="738" alt="image" src="https://github.com/user-attachments/assets/b057de05-d752-49f6-b1a9-51680ae44504" />
<img width="1903" height="330" alt="image" src="https://github.com/user-attachments/assets/fbbde564-e01c-4577-9376-c646f6e7034b" />

The call on the method _update_picking_from_group_key is the method that is messing it up as it updates the previous picking. It actually only updates the previous picking/move. But later in the execution, the picking/move is also updated, so the call on the method is not necessary. 

Also, this method is called by _purchase_location_by_line_ and neither _purchase_delivery_split_date_ nor _purchase_location_by_line_ misses this method as it's not called in any other place (at least I did not found any other case) and the tests are still working fine.

If the changes are approved and merged, I'll remove the hole method _update_picking_from_group_key from both appearances.

DavidJForgeFlow committed:
[FIX] purchase_delivery_split_date

When the PO is already confirmed and you add a new line with a different date than the existing one, the new receipt is created but the old receipt and move have the dates updated with no reason. The call on the method _update_picking_from_group_key is the method that is messing it up as it updates the previous picking. Also ensures that date is updated in the stock.moves when the value changes in purchase.order.line

LoisRForgeFlow commented:
The case that is now not covered is updating the date of an existing line:

<img width="1261" height="668" alt="image" src="https://github.com/user-attachments/assets/fc715f23-a601-4f4b-9fde-cb2c610c3cf1" />


I think a test case for each scenario should be added:
1. update existing line date.
2. create a new line.

DavidJForgeFlow committed:
[FIX] purchase_location_by_line

Remove the method _update_picking_from_group_key as it is removed in the parent module.

LoisRForgeFlow reviewed (approved):
Tested, it works good now. Thanks for the test cases.

LoisRForgeFlow commented:
@OCA/procurement-purchase-maintainers could you have a look?

HviorForgeFlow commented:
/ocabot merge patch

LoisRForgeFlow commented:
@DavidJForgeFlow could you fwport this?

DavidJForgeFlow commented:
Forward ports done!
